packages/frameworks-nuxt/src/runtime/lib/server.ts

```diff
@@ -0,0 +1,80 @@
+import { Auth, setEnvDefaults } from "@auth/core"
+import type { AuthConfig } from "@auth/core"
+import { getRequestFromEvent, getBasePath } from "../utils"
+import { defineEventHandler } from "h3"
+import type { H3Event } from "h3"
+import type { Session } from "@auth/core/types"
+
+/**
+ * Handle auth calls in Nuxt project.
+ *
+ *  @example
+ * ```ts title="server/routes/auth/[...].ts"
+ * import { NuxtAuthHandler } from "#auth"
+ * import GitHub from "@auth/nuxt/providers/github"
+ *
+ * export default defineEventHandler(async (event) => {
+ *    return NuxtAuthHandler({ providers: [GitHub] })
+ * })
+ * ```
+ *
+ * @param config AuthConfig
+ * @returns EventHandler
+ */
+export function NuxtAuthHandler(config: AuthConfig) {
+  return defineEventHandler(async (event) => {
+    /**
+     * If the request is a prerender request, do nothing.
+     */
+    if (
+      event.node.req.headers?.["x-nitro-prerender"] &&
+      import.meta.env.NODE_ENV === "prerender"
+    ) {
+      return
+    }
+
+    // Do not handle source maps
+    if (event.node.req.url?.includes(".js.map")) return
+
+    // Set missing items/ override items based on the environment variables
+    setEnvDefaults(process.env, config)
+
+    const request = await getRequestFromEvent(event)
+    config.trustHost ??= true
+    config.basePath ??= getBasePath(request)
+
+    return await Auth(request, config)
+  })
+}
+
+/**
+ * Get and returns the session to be used in server side, like in API routes.
+ *
+ * @example
+ * ```ts title="server/api/protected.get.ts"
+ * import { getServerSession } from "#auth"
+ *
+ * export default defineEventHandler(async (event) => {
+ *  const session = await getServerSession(event)
+ *  if (!session) return null
+ *
+ *  return {
+ *    message: "Protected data",
+ *  }
+ * })
+ * ```
+ *
+ * @param event H3Event
+ * @returns Session
+ */
+export async function getServerSession(
+  event: H3Event
+): Promise<Session | null> {
+  const headers = event.headers
+  const data = await $fetch<Session>("/auth/session", {
```

Comment:
Shouldn't this be `${config.basePath}/session`

I can see that having to pass in config as an argument everytime one of the utility functions is called could be annoying

My suggestion would be to change the signature to be 
```ts
async function getServerSession(event: H3Event, config: authConfig)
// ^ not exported
```

and do the same to any other utilities that will be written

after that using the function below

```ts
export function generateServerUtils(config: authConfig) {
  return {
    getServerSession: async (event: H3Event) => await getServerSession(event, config),
  }
}
```

the consumer can generate their own utilities in `~/server/utils/` folder which will then be auto imported by nuxt without any further assistance.
see -> [auto-imports](https://nuxt.com/docs/guide/concepts/auto-imports) and [server-utilities](https://nuxt.com/docs/guide/directory-structure/server#server-utilities)

```ts
// ~/server/utils/auth.ts
import { generateServerUtils } from "#auth"
import { authConfig } from "@/server/[api/auth/, routes/auth/, routes/whatever][...].ts"

export const { getServerSession } = generateServerUtils(authConfig );

```

Reply:
Thanks for the review, I'll get to it over the weekend